### PR TITLE
Include '@types/node' in ATA if commonjs modules are used

### PR DESCRIPTION
**Pull request by andy-ms**
@DanielRosenwasser had recommended doing this.

### CODE REVIEW

File: src/compiler/types.ts [hunk 1]
```diff
@@ -3968,6 +3968,7 @@ namespace ts {
         typeAcquisition: TypeAcquisition;               // Used to customize the type acquisition process
         compilerOptions: CompilerOptions;               // Used as a source for typing inference
         unresolvedImports: ReadonlyArray<string>;       // List of unresolved module ids from imports
+        containsCommonJsRequire: boolean;               // True if there is a `require()` call somewhere in the program
```
Comment (andy-ms) [inline src/compiler/types.ts]: Should this be made optional? This interface is casted to in one place but I can't find anywhere we actually create it -- but it does happen to share a shape with `interface DiscoverTypings` in `server.ts` so maybe it comes from there?

**Comment by DanielRosenwasser** [comment]
I just realized that there's one place this could cause problems, which is in a `webpack.config.js`, which apparently has its own type for `require`. I wonder if that's still okay?

**Comment by andy-ms** [comment]
@DanielRosenwasser What do those `require` calls look like? Unless there's one argument and it's a string literal, we shouldn't count it as a commonjs `require`.

**Comment by mhegazy** [comment]
I do not think this the right fix. first it is a bit of a heavy hammer for the `require` missing issue to include all of node. I think for this scenario we can just add something to suppress the error for `require` specifically in some contexts..

the other issue however is that does not handle the scenario listed in #21164, with the flashing of the error untill ATA gets to run. i think this may be different than the original problem here, but it may be related too..

let's talk about the approach we can take on Tuesday.

**Comment by mhegazy** [comment]
Closing in favor of the proposal in https://github.com/Microsoft/TypeScript/issues/21933

**Pull request closed without merging**